server/src/core/distribution/run-distribution.ts

```diff
@@ -46,11 +47,19 @@ interface CreateDistributionPlanResult {
   beneficiariesSummaries:DistributionPlanBeneficiariesSummaries;
 }
 
-export async function runDonationDistribution(db: Db, args: DonationDistributionArgs): Promise<DonationDistributionEvent[]> {
+export async function runDonationDistribution(db: Db, args: DonationDistributionArgs, filter: BeneficiaryFilter): Promise<DonationDistributionEvent[]> {
   const { periodLength, periodLimit, users } = args;
 
-  const beneficiaries = await findEligibleBeneficiaries(db, periodLimit, periodLength);
-  const donorIds = beneficiaries.reduce<string[]>((acc, b) => [...acc, ...b.donors], []);
+  const beneficiaries = await findEligibleBeneficiaries(db, periodLimit, periodLength, filter);
+  let donorIds;
+  if (beneficiaries.length && beneficiaries[0].donors) {
+    donorIds = beneficiaries.reduce<string[]>((acc, b) => [...acc, ...b.donors], []);
```

Comment:
Because `b.donors` is not possibly `undefined`, you should update the reduce to:
```js
reduce((acc, b) => [...acc, ...(b.donors || [])], [])
```
hopefully that's still readable. But basically defaults to an empty array if `b.donors` doesn't exist.